pac4j-core/src/main/java/org/pac4j/core/client/Clients.java

```diff
@@ -73,28 +74,32 @@ public Clients(final Client... clients) {
     }
 
     /**
-     * Initialize all clients by computing callback urls.
+     * Initialize all clients by computing callback urls if necessary.
      */
     @Override
     protected void internalInit() {
         CommonHelper.assertNotNull("clients", this.clients);
-        if (CommonHelper.isNotBlank(this.callbackUrl)) {
-            for (final Client client : this.clients) {
-                if (client instanceof IndirectClient) {
-                    final IndirectClient indirectClient = (IndirectClient) client;
-                    String indirectClientCallbackUrl = indirectClient.getCallbackUrl();
-                    // no callback url defined for the client -> set it with the group callback url
-                    if (indirectClientCallbackUrl == null) {
-                        indirectClient.setCallbackUrl(this.callbackUrl);
-                        indirectClientCallbackUrl = this.callbackUrl;
-                    }
-                    // if the "clientName" parameter is not already part of the callback url, add it unless the client
-                    // has indicated to not include it.
-                    if (indirectClient.isIncludeClientNameInCallbackUrl() &&
-                            indirectClientCallbackUrl.indexOf(this.clientNameParameter + "=") < 0) {
-                        indirectClient.setCallbackUrl(CommonHelper.addParameter(indirectClientCallbackUrl, this.clientNameParameter,
-                                indirectClient.getName()));
-                    }
+        final HashSet<String> names = new HashSet<>();
+        for (final Client client : this.clients) {
+            final String name = client.getName();
+            if (names.contains(name)) {
```

Comment:
`Clients` gets configured somewhere right? These class names that get put into Clients are just Strings? 

My idea is to ensure that Clients prevents duplicates as opposed to manually doing it here.